content/docs/learning-guide/tutorials/custom-tools/shape-example-py.md

```diff
@@ -0,0 +1,470 @@
+description: Learn to create a custom shape tool in Python to extend Open 3D Engine (O3DE).
+weight: 300
+toc: true
+---
+
+In this tutorial, you'll create a custom tool, **Shape Example**, written in Python that extends **Open 3D Engine (O3DE) Editor** . The Shape Example tool allows you to create entities with a Shape component and configure their component properties. You'll learn how to use the `PythonToolGem` template and practice Python development with O3DE's Tools UI API and other O3DE APIs.
+
+![An image of the Shape Example tool and some entities created by it.](/images/learning-guide/tutorials/custom-tools/shape-example/python-shape-example-demo.png)
+
+This tutorial is based off of the **PyShapeExample** Gem in [`o3de/sample-code-gems` repository](https://github.com/o3de/sample-code-gems/tree/main/py_gems/PyShapeExample). You can reference the PyShapeExample Gem sample as you follow along this tutorial.
+
+By the end of this tutorial, you'll be able to create your own tools that extends O3DE Editor.
+
+
+## Create a tool with the `PythonToolGem` template
+
+The `PythonToolGem` template contains a basic Python framework to create a dialog window that extends the O3DE Editor.
+
+1. Create a Gem by using the **O3DE CLI (`o3de`)** script that's in your engine source directory. Specify the name `PyShapeExample` by using the `--gem-name` option and the template `PythonToolGem` by using the `--template-name` option. By default, this creates the Gem directory in `<user>/.o3de/Gems` -- or you can specify a location by using the `--gem-path` option. Depending on the location, this automatically registers the Gem to either the `.o3de` registry, engine, or project.
+
+    ```cmd
+    scripts/o3de create-gem --gem-name ShapeExample --template-name PyToolGem
+    ```
+
+2. (Optional) Register the Gem to your project. This step is optional because when you create a Gem using the O3DE CLI in the previous step, it automatically registers the Gem.
+
+    ```cmd
+    scripts/o3de register -gp <gem-path> -espp <project-path>
+    ```
+
+3. Add the Gem in your project by using the `o3de` script.
+
+    ```cmd
+    scripts/o3de enable-gem -gn PythonToolGem -pp <project-path>
+    ```
+
+    Or, enable the Gem using the Project Manager (refer to [Adding and Removing Gems in a Project](/docs/user-guide/project-config/add-remove-gems.md)).
+
+4. Build the project by using the `o3de` script (refer to [Build a project](https://o3de.org/docs/user-guide/build/configure-and-build/#build-a-project)). Or, use the Project Manager (refer to the **Build** action in the [Project Manager](https://o3de.org/docs/user-guide/project-config/project-manager/)) page.
+
+5. Open O3DE Editor for your project.
+
+6. Open the **Python Scripts** panel by selecting **Tools > Other > Python Scripts** from the file menu. (See A in the following image.)
+
+7. Open `pyshapeexample_dialog` in the Python Scripts panel. (See B in the following image.)
+
```

Comment:
Discussed this offline, this was a bug due to the fact that our shape example (and `PythonToolGem` template) weren't setting the `QtForPython` gem as a dependency correctly, so it wasn't being bootstrapped properly. This section can be updated to match the C++ version, where it is properly registered in the "Tools -> Examples" menu and has a toolbar icon.

We *should* move these details though down to the debugging section because it is great for quick iteration.